detox/test/android/app/src/main/java/com/example/utils/DetoxSoLoader.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.utils
+
+import android.content.Context
+import com.facebook.react.modules.systeminfo.ReactNativeVersion
+import com.facebook.soloader.SoLoader
+
+private const val OpenSourceMergedSoMappingClassName = "com.facebook.react.soloader.OpenSourceMergedSoMapping"
+private const val ExternalSoMappingClassName ="com.facebook.soloader.ExternalSoMapping"
+
+object DetoxSoLoader {
+    fun init(appContext: Context) {
+        if (ReactNativeVersion.VERSION.run { get("minor") as Int } >= 76) {
+            val mergedSoMappingClass = Class.forName(OpenSourceMergedSoMappingClassName)
+            val externalSoMapping = Class.forName(ExternalSoMappingClassName)
+            val initMethod = SoLoader::class.java.getMethod("init", Context::class.java, externalSoMapping)
```

Comment:
Interesting. How come `SoLoader.init(context, boolean)` can no longer be used? Is it because native modules' `.so` files need to be dynamically loaded now? If so, then maybe this change belongs in the new-arch branch? 🤔

Reply:
No, this is a part of rn76 setup. We use SoLoader.init(context, boolean) in 75 and the second overload in 76. It is also mandatory for the new arch. Any way, this is aligned with the react native template